test/CancellationHandleTest.cpp

```diff
@@ -285,10 +287,31 @@ TEST(CancellationHandle, verifyCheckAfterDeadlineMissDoesReportProperly) {
             ContainsRegex("by [0-9]ms")));
 }
 
+// _____________________________________________________________________________
+
+TEST(CancellationHandle, expectDisabledHandleIsAlwaysFalse) {
+  CancellationHandle<DISABLED> handle;
+
+  EXPECT_FALSE(handle.isCancelled());
+  EXPECT_NO_THROW(handle.throwIfCancelled("Abc"));
+}
+
 // Make sure member functions still exist when no watch dog functionality
 // is available to make the code simpler. In this case the functions should
 // be no-op.
 static_assert(std::is_member_function_pointer_v<
-              decltype(&CancellationHandle<false>::startWatchDog)>);
+              decltype(&CancellationHandle<NO_WATCH_DOG>::startWatchDog)>);
+static_assert(std::is_member_function_pointer_v<
+              decltype(&CancellationHandle<NO_WATCH_DOG>::resetWatchDogState)>);
+static_assert(std::is_member_function_pointer_v<
+              decltype(&CancellationHandle<DISABLED>::startWatchDog)>);
+static_assert(std::is_member_function_pointer_v<
```

Comment:
Factor out the `is_member_function...` and the decltype, 
s.t. you only need to write

`assertMemFun(&CancellationHandle<BLA>::startWatchDog)`;